system-addon/lib/ASRouter.jsm

```diff
@@ -78,21 +162,54 @@ class _ASRouter {
     throw new Error("Do not modify this.state directy. Instead, call this.setState(newState)");
   }
 
+  /**
+   * loadMessagesFromAllProviders - Loads messages from all providers if they require updates.
+   *                                Checks the .lastUpdated field on each provider to see if updates are needed
+   * @memberof _ASRouter
+   */
+  async loadMessagesFromAllProviders() {
+    const needsUpdate = this.state.providers.filter(provider => MessageLoaderUtils.shouldProviderUpdate(provider));
+    // Don't do extra work if we don't need any updates
+    if (needsUpdate.length) {
+      let newState = {messages: [], providers: []};
+      for (const provider of this.state.providers) {
```

Comment:
I still need to iterate over the non-updated providers because I have to add their messages back in to the new state. I could change this to `needsUpdate.includes(provider)` array instead of re-running the function though